src/uwtools/tests/test_schemas.py

```diff
@@ -456,10 +455,116 @@ def test_schema_rocoto_workflow_cycledef():
 # sfc-climo-gen
 
 
+@fixture
+def sfc_climo_gen_fcstprop():
+    return partial(schema_validator, "sfc-climo-gen", "properties", "sfc_climo_gen", "properties")
```

Comment:
I'm confused here by "sfc-climo-gen" and "properties" showing up twice when being passed to `partial`. 

I looked up `partial` as a python function that accepts arguments and it looks to me through some articles online that not all the arguments being passed to the call have a fixed result yet (still in the process of collecting the value for the variable)?

I think these values are potential placeholders for the call to `schema_validator` and just wanted some clarification on it.

Reply:
That was my understanding of `partial`, basically leaving "properties" as a preset for later additions to not have to specify it

Reply:
Python's `partial()` is an implementation of the functional programming concept called [partial application](https://en.wikipedia.org/wiki/Partial_application), which turns a function of n arguments into a function of n-m arguments by providing a fixed value for the m arguments. The benefit is that you get a new function with a (hopefully) more specific name that you can call without repeatedly supplying the same fixed argument.

The arguments to `partial()` are 1. The name of the n-argument function that is being simplified, and 2. The m arguments that we want to supply fixed values for.

In this case, we're partially applying (also sometimes called [currying](https://en.wikipedia.org/wiki/Currying) after the mathematician Haskell Curry -- after whom the programming language Haskell is also named -- though I think there are some subtle technical differences between partial application and currying) _all_ the arguments of the `schema_validator` function: First we supply the base name of the `.jsonschema` file to use for validation (which is why we see `sfc-climo-gen` with hyphens here, which will be used to locate the file `sfc-climo-gen.jsonschema`), then we supply a series of JSON keys used to navigate from the top of the schema to the block that we're interested in testing -- that's why we see `sfc_climo_gen` with underscores -- that's the first JSON key in `sfc-climo-gen.jsonschema`).

So, this whole function returns a _new function_ that is a schema validator for a specific part of the whole `sfc_climo_gen` schema. It might help to have a look at the `schema_validator()` function, too.

Reply:
Thanks for the information, this definitely helps clarify the use of `partial` and what exactly is happening on this line.

I previously took a look at the `schema_validator()` function which helped with some understanding. I think the big connection I was missing here was the subtle difference between `sfc-climo-gen` and `sfc_climo_gen`. This makes more sense that it's looking for a specific key and navigating further into the schema with the underscores vs. the other use case with hyphens.